artiq/frontend/artiq_coremgmt.py

```diff
@@ -85,6 +89,24 @@ def get_argparser():
     t_boot = tools.add_parser("reboot",
                               help="reboot the running system")
 
+    # flashing
+    t_flash = tools.add_parser("flash",
+                               help="flash the running system")
+
+    p_zynq = t_flash.add_argument("-z", "--zynq", default=False,
```

Comment:
any chance for auto detection?

Reply:
I guess we can simply try both riscv and zynq binaries, then go for the one that succeed.

Reply:
Just like runtime/satman is currently detected.